sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CfgSerializer.cs

```diff
@@ -0,0 +1,147 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.SymbolicExecution.ControlFlowGraph;
+
+namespace SonarAnalyzer.Helpers
+{
+    internal static class CfgSerializer
+    {
+        public static string Serialize(string methodName, IControlFlowGraph cfg)
+        {
+            var stringBuilder = new StringBuilder();
+            using (var writer = new StringWriter(stringBuilder))
+            {
+                Serialize(methodName, cfg, writer);
+            }
+            return stringBuilder.ToString();
+        }
+
+        public static void Serialize(string methodName, IControlFlowGraph cfg, TextWriter writer)
+        {
+            new CfgWalker(new DotWriter(writer)).Visit(methodName, cfg);
+        }
+
+        private class CfgWalker
+        {
+            private readonly BlockIdMap blockId = new BlockIdMap();
+            private readonly DotWriter writer;
+
+            public CfgWalker(DotWriter writer)
+            {
+                this.writer = writer;
+            }
+
+            public void Visit(string methodName, IControlFlowGraph cfg)
+            {
+                writer.WriteGraphStart(methodName);
+
+                foreach (var block in cfg.Blocks)
+                {
+                    Visit(block);
+                }
+
+                writer.WriteGraphEnd();
+            }
+
+            private void Visit(Block block)
+            {
+                Func<Block, string> getLabel = b => string.Empty;
+
+                if (block is BinaryBranchBlock binaryBranchBlock)
```

Comment:
Minor: could pattern matching in C#7 be used here instead i.e. a _switch_ instead of multiple _if_s?